### PR TITLE
DEV-1760 | Better logging for magic link flow

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

This PR is initially into DEV-1372 because it pre-supposes those changes. After that PR has been merged, base on this one should be set to `main`.

#### What's this PR do?

Adds info level logs for all code involved in magic link request cycle.

Additionally, I found that the call to `send_templated_email` in `apps.api.views.RequestContributorTokenEmailView.post` was happening synchronously, when it should be delayed. I changed that and updated some tests accordingly.

#### Why are we doing this? How does it help us?

Better dev experience.

#### How should this be manually tested? Please include detailed step-by-step instructions.

N/A

#### Have automated unit tests been added? If not, why?

No. Not going to test for logging.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-1760](https://news-revenue-hub.atlassian.net/jira/software/c/projects/DEV/boards/3?modal=detail&selectedIssue=DEV-1760)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

N/A

### CODE REVIEW

File: apps/api/views.py [hunk 1]
```diff
@@ -144,14 +146,21 @@ class RequestContributorTokenEmailView(APIView):
     throttle_classes = [ContributorRateThrottle]
 
     def post(self, request, *args, **kwargs):
+        logger.info("[RequestContributorTokenEmailView] Request received for magic link %s", request.data)
```
Comment (benjaminEwhite) [inline apps/api/views.py:149]:
```suggestion
        logger.info("[RequestContributorTokenEmailView][post] Request received for magic link %s", request.data)
```

File: apps/api/views.py [hunk 2]
```diff
@@ -144,14 +146,21 @@ class RequestContributorTokenEmailView(APIView):
...
+            logger.info(
+                "[RequestContributorTokenEmailView] Created new contributor with email %s", request.data["email"]
+            )
+
         serializer.update_short_lived_token(contributor)
 
         domain = _construct_rp_domain(serializer.data.get("subdomain", ""), request.headers.get("Referer", ""))
 
         if not domain:
+            logger.info("[RequestContributorTokenEmailView] Could not determine domain for request")
```
Comment (benjaminEwhite) [inline apps/api/views.py:163]:
```suggestion
            logger.info("[RequestContributorTokenEmailView][post] Could not determine domain for request")
```

File: apps/api/views.py [hunk 3]
```diff
@@ -184,6 +193,7 @@ class VerifyContributorTokenView(APIView):
     permission_classes = []
 
     def post(self, request, *args, **kwargs):
+        logger.info("[VerifyContributorTokenView] Request received for user (%s)", request.user)
```
Comment (benjaminEwhite) [inline apps/api/views.py:196]:
```suggestion
        logger.info("[VerifyContributorTokenView][post] Request received for user (%s)", request.user)
```

**Review by x110dc — APPROVED** [review]

File: apps/api/views.py [hunk 4]
```diff
@@ -38,6 +35,7 @@ def _construct_rp_domain(subdomain, referer):
 
     Return RP specific domain or None if not found.
     """
+    logger.info("[_construct_rp_domain] constructing rp domain for subdomain (%s)and referer (%s)", subdomain, referer)
```
Comment (njharman) [inline apps/api/views.py:38]: zomg! missing space before "and".

**Review by njharman — APPROVED** [review]
lgtm, needs update from main.

I'm not sure I'm looking at right branch.  But I get error trying to run ./manage.py No module Django.utils...

**Review by x110dc — APPROVED** [review]

**Pull request merged**